src/NServiceBus.Core/Routing/RoutingSettings.cs

```diff
@@ -23,6 +23,9 @@ internal RoutingSettings(SettingsHolder settings)
         /// <param name="destination">Destination endpoint.</param>
         public void RouteToEndpoint(Type messageType, string destination)
         {
+            if(destination.Contains("@"))
+                throw new ArgumentException($"Expected an endpoint name but received '{destination}'. Use routing file to specify physical address of the endpoint.");
```

Comment:
Is

> Use routing file to specify physical address of the endpoint.

correct? If I understand correctly the file based routing was moved to MSMQ only